compiler/parser/RdrHsSyn.hs

```diff
@@ -3018,9 +3018,9 @@ mkChoiceByKeyInstanceDecl templateName ValidTemplate{..} CombinedChoiceData { cc
 mkInterfaceFixedChoiceInstanceDecl :: Located RdrName -> InterfaceChoiceSignature -> [LHsDecl GhcPs]
 mkInterfaceFixedChoiceInstanceDecl tycon InterfaceChoiceSignature {..} =
   [ mkInstance "HasToAnyChoice"
-      (mkTemplateClassMethod "_toAnyChoice" [proxy] (mkPrimitive "primitive" "EToAnyInterfaceChoice" `mkApp` nil ifaceType) Nothing)
+      (mkTemplateClassMethod "_toAnyChoice" [proxy] (mkPrimitive "primitive" "EToAnyChoice" `mkApp` nil ifaceType) Nothing)
```

Comment:
Unlike `EToAnyInterfaceChoice`, the definition of `EToAnyChoice` doesn't inspect the type of the proxy argument, so now we can go point-free

```suggestion
      (mkTemplateClassMethod "_toAnyChoice" [] (mkPrimitive "primitive" "EToAnyChoice") Nothing)
```

This will change the generated code:

```diff
-   _toAnyChoice _ = GHC.Types.primitive @"EToAnyChoice" ([] : [MyInterface])
+   _toAnyChoice = GHC.Types.primitive @"EToAnyChoice"
```